wordpress-mobile/WordPress-iOS
# Limit video uploads on free sites

This PR prevents videos longer than 5 minutes to be uploaded on free sites. This applies to captured or selected videos from:

- My Site - Media
- Editor (posts and pages)
- Stories

Notes:
- Video upload from "Other Apps" was already disabled, for videos of any duration, before this PR
- On Stories, there was an existing video limit of 29 seconds: this has not been changed

Fixes #NA

To test:

Pre requisites: 
- make sure to have local files both on device and from other apps (e.g Files) videos that are longer than 5 minutes. 
- Tip: to use shorter videos (this is particularly useful when testing the capture), you can temporary tweak the following code (`WPMediaAsset+VideoLimits.swift`)
    `let maximumVideoDurationForFreeSites: CGFloat = 300`
  choosing a shorter value than 5 minutes (e.g. 5 seconds)


1. Go to My site and make sure to select a site with a free plan
2. Go to Media
3. Tap on the + (top right corner)
4. Tap on "Take Photo or Video"
5. Record a video longer that the maximum allowed duration
6. Tap on "Use video"
7. Make sure the "Video not uploaded" (see screenshot below) alert is shown, and no video is added to the media library
8. Record a video shorter than the maximum allowed duration, and make sure it can be uploaded instead.
9. Tap on + again
10. Tap on "Choose from My Device"
11. Make sure that videos longer than the maximum allowed duration are greyed out and cannot be selected. Also, make sure that the "Selection not allowed" alert appears (see screenshot below)
12. Optional (this was not changed in this PR): tap on "Other apps", and make sure that video files (of any duration) are still greyed out and cannot be uploaded.
13. Go back and tap on the blue Floating Action Button (FAB)
14. Select "Story Post"
15. Tap on the  icon in the lower left corner and repeat step 11
16. Optional (this was not changed in this PR): record a story video and make sure the max duration of 29 seconds applies
17. Go back and either tap again on the FAB and then "Blog Post" to create a new post, or edit an existing post from the "Posts" section
18. Repeat steps 4 to 12 (you can either add multiple video blocks or tap "Add Video" on a video block that has no media yet, to display the selection options)
19. Go back and either create a new page from the FAB or edit an existing one from the "Pages" section
20. Repeat step 18
21. Repeat the steps 2 to 20 on a paid site, and make sure you can upload videos of any duration

Captured video too long | Selected video too long | Choose video
-- | -- | --
<img width=300 src=https://user-images.githubusercontent.com/34376330/146723022-e3710621-624a-41ca-83d6-e76b3ee76750.png> | <img width=300 src=https://user-images.githubusercontent.com/34376330/146723079-7b71b861-6d49-496a-b8c7-d38000c1bc23.png> | !<img width=300 src=https://user-images.githubusercontent.com/34376330/146723142-a4359df7-6e94-4707-b03f-7946467397ff.png>





## Regression Notes
1. Potential unintended areas of impact
Limited impact: these changes impact the video upload sections in media, Gutenberg and Stories

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Test all the impacted cases, both with free and paid plans

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
